internal/eventstore/repository/sql/query.go

```diff
@@ -125,31 +123,36 @@ func eventsScanner(scanner scan, dest interface{}) (err error) {
 	return nil
 }
 
-func prepareCondition(criteria querier, filters []*repository.Filter) (clause string, values []interface{}) {
-	values = make([]interface{}, len(filters))
-	clauses := make([]string, len(filters))
+func prepareCondition(criteria querier, filters [][]*repository.Filter) (clause string, values []interface{}) {
+	values = make([]interface{}, 0, len(filters))
 
 	if len(filters) == 0 {
 		return clause, values
 	}
-	for i, filter := range filters {
-		value := filter.Value
-		switch value.(type) {
-		case []bool, []float64, []int64, []string, []repository.AggregateType, []repository.EventType, *[]bool, *[]float64, *[]int64, *[]string, *[]repository.AggregateType, *[]repository.EventType:
-			value = pq.Array(value)
-		case map[string]interface{}:
-			var err error
-			value, err = json.Marshal(value)
-			logging.Log("SQL-BSsNy").OnError(err).Warn("unable to marshal search value")
-		}
 
-		clauses[i] = getCondition(criteria, filter)
-		if clauses[i] == "" {
-			return "", nil
+	clauses := make([]string, len(filters))
+	for idx, filter := range filters {
+		subClauses := make([]string, len(filter))
+		for subIdx, f := range filter {
+			value := f.Value
+			switch value.(type) {
+			case []bool, []float64, []int64, []string, []repository.AggregateType, []repository.EventType, *[]bool, *[]float64, *[]int64, *[]string, *[]repository.AggregateType, *[]repository.EventType:
+				value = pq.Array(value)
+			case map[string]interface{}:
+				var err error
+				value, err = json.Marshal(value)
+				logging.Log("SQL-BSsNy").OnError(err).Warn("unable to marshal search value")
```

Comment:
i know this part is unchanged, but shouldn't we ignore the filter on error?

Reply:
the clause will be ignored now